### ISSUE TITLE
Stuck on CsvRelation.scala:249

### ISSUE THREAD

**Issue by IoDmitri** [databricks/spark-csv#260]
[Here](http://stackoverflow.com/questions/35330067/spark-job-stuck-on-take-at-csvrelation-scala249) is an open SO question that I posted. The issue Im noticing is that while running on Amazon EMR, the Spark client is stuck on CsvRelation.

**Comment by HyukjinKwon** [comment]
@IoDmitri Would you please give some details here rather than just a link (such as the error message and your data format (or a sample of your data))?

**Comment by IoDmitri** [comment]
@HyukjinKwon  certainly, [here](https://s3-us-west-2.amazonaws.com/appdataforme/formatted.csv) is the data. As for the error, that the thing there is no error, its that the Spark UI shows CsvRelation.scala:249 as a task, but its at 0%, and its been that way for an hour. Imagine [this](http://imgur.com/HngNmaz) image, but instead of ``take at CsvRelation.Scala`` being completed, its not completed and the duration is now at an hour. I do apologize for my lack of clarity on this issue, if you need more data/information please dont hesitate to let me know.

**Comment by pinifaran** [comment]
We had a similar problem (stuck in data bricks exactly on the same line), and the RC of the problem was the fact we are running with yarn client configuration (probably due to MapR bug when running via oozie Spark job in yarn client configuration).

When we switched to working in yarn master configuration the problem was resolved.
